src/dialogsearch.cpp

```diff
@@ -144,7 +144,21 @@ void DialogSearch::on_button_search_clicked()
     }
     else if (ui->radio_switch->isChecked())
     {
-        //TODO implement me
+        int switchID = ui->combo_switch->currentData().toInt();
+
+        search_predicate = [switchID](const RPG::EventCommand& com)
+        {
+            switch(com.code)
+            {
+                case Cmd::ControlSwitches:
+                    return (com.parameters[0] != 2 && com.parameters[1] == switchID) ||
+                        (com.parameters[0] == 1 && com.parameters[2] == switchID);
+                case Cmd::ConditionalBranch:
+                    return com.parameters[0] == 0 && com.parameters[1] == switchID;
+                default:
+                    return false;
```

Comment:
By looking at it: No, this looks at event commands. And battle events are not scanned but that would be an additional bonus.

But there is one case that could be implemented with not too much work: 
When ```com.parameter[0]``` is 2 then parameter 1 and 2 define a switch range (applies to variables, too)